staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go

```diff
@@ -309,112 +339,155 @@ func TestCustomResourceValidationErrors(t *testing.T) {
+		} else {
+			validationSchema = noxuDefinition.Spec.Versions[i].Schema
+		}
 
-	// set stricter schema
-	noxuDefinition.Spec.Validation.OpenAPIV3Schema.Required = []string{"alpha", "beta", "epsilon"}
+		// set stricter schema
+		validationSchema.Required = []string{"alpha", "beta", "epsilon"}
 
-	noxuDefinition, err = fixtures.CreateNewCustomResourceDefinition(noxuDefinition, apiExtensionClient, dynamicClient)
-	if err != nil {
-		t.Fatal(err)
-	}
-	ns := "not-the-default"
-	noxuResourceClient := newNamespacedCustomResourceClient(ns, dynamicClient, noxuDefinition)
+		noxuDefinition, err = fixtures.CreateNewCustomResourceDefinition(noxuDefinition, apiExtensionClient, dynamicClient)
+		if err != nil {
+			t.Fatal(err)
+		}
+		ns := "not-the-default"
 
-	// CR is rejected
-	_, err = instantiateCustomResource(t, newNoxuValidationInstance(ns, "foo"), noxuResourceClient, noxuDefinition)
-	if err == nil {
-		t.Fatalf("unexpected non-error: CR should be rejected")
-	}
+		noxuResourceClient := newNamespacedCustomResourceVersionedClient(ns, dynamicClient, noxuDefinition, v.Name)
+		instanceToCreate := newNoxuValidationInstance(ns, "foo")
+		instanceToCreate.Object["apiVersion"] = fmt.Sprintf("%s/%s", noxuDefinition.Spec.Group, v.Name)
 
-	// update the CRD to a less stricter schema
-	_, err = updateCustomResourceDefinitionWithRetry(apiExtensionClient, "noxus.mygroup.example.com", func(crd *apiextensionsv1beta1.CustomResourceDefinition) {
-		crd.Spec.Validation.OpenAPIV3Schema.Required = []string{"alpha", "beta"}
-	})
-	if err != nil {
-		t.Fatal(err)
-	}
+		// CR is rejected
+		_, err = instantiateVersionedCustomResource(t, instanceToCreate, noxuResourceClient, noxuDefinition, v.Name)
+		if err == nil {
+			t.Fatalf("unexpected non-error: CR should be rejected")
+		}
+
+		// update the CRD to a less stricter schema
+		_, err = updateCustomResourceDefinitionWithRetry(apiExtensionClient, "noxus.mygroup.example.com", func(crd *apiextensionsv1beta1.CustomResourceDefinition) {
+			if i == 0 {
+				crd.Spec.Validation.OpenAPIV3Schema.Required = []string{"alpha", "beta"}
+			} else {
+				crd.Spec.Versions[1].Schema.Required = []string{"alpha", "beta"}
+			}
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
 
-	// CR is now accepted
-	err = wait.Poll(500*time.Millisecond, wait.ForeverTestTimeout, func() (bool, error) {
-		_, err := noxuResourceClient.Create(newNoxuValidationInstance(ns, "foo"), metav1.CreateOptions{})
-		if statusError, isStatus := err.(*apierrors.StatusError); isStatus {
-			if strings.Contains(statusError.Error(), "is invalid") {
-				return false, nil
+		// CR is now accepted
+		err = wait.Poll(500*time.Millisecond, wait.ForeverTestTimeout, func() (bool, error) {
+			_, err := noxuResourceClient.Create(instanceToCreate, metav1.CreateOptions{})
+			if statusError, isStatus := err.(*apierrors.StatusError); isStatus {
+				if strings.Contains(statusError.Error(), "is invalid") {
```

Comment:
don't we have `IsInvalid`? Also the http code should help.

Reply:
This line is unchanged in this PR (probably the tests can be better reviewed with [indentation changes hidden](https://github.com/kubernetes/kubernetes/pull/67553/files?utf8=%E2%9C%93&diff=split&w=1#diff-a1418db843e532f56ceee297e82cf26eR412))

I can do the fix in this PR if you want